jax/_src/prng.py

```diff
@@ -317,6 +318,10 @@ def transpose(self, *_, **__) -> PRNGKeyArray: assert False
     'squeeze', 'swapaxes', 'take', 'transpose', 'T'])
 basearray.Array.register(PRNGKeyArrayImpl)
 
+def zeros_like_prngkey(x: PRNGKeyArrayImpl) -> PRNGKeyArrayImpl:
+  return PRNGKeyArrayImpl(x.impl, jnp.zeros_like(x._base_array))
```

Comment:
Should we make `jnp.zeros_like` work on key arrays (as discussed elsewhere) and then implement this on top of that directly?

Reply:
Yeah, that's a good idea – though did we make a final decision on `zeros_like`?

Reply:
For both `zeros` and `zeros_like`, the latest thought was "a key array wrapping zeros but considered already-consumed by the reuse checker" (understanding that the latter does not yet exist).

Reply:
And we could possibly treat `empty` similarly, if needed.

Reply:
OK, I'll do that first and then rewrite this PR on top of it.

Reply:
Done, see #16033